source/extensions/filters/http/aws_lambda/config.cc

```diff
@@ -194,12 +194,15 @@ void AWSLambdaConfigImpl::AWSLambdaStsRefresher::init(Event::Dispatcher &dispatc
     } else {
         ENVOY_LOG(debug, "{}: STS enabled without time based refresh",__func__);
     }
-    file_watcher_->addWatch(
+    // the result of this function MUST NOT be ignored (see [[nodiscard]]) - but
+    // what should we do with it?? TODO
+    RETURN_IF_NOT_OK(file_watcher_->addWatch(
```

Comment:
Shit. Fair point that this function should not return a value. Consider instead if we just add a log saying that we should never reach this due to a always ok return?

Reply:
there's also `THROW_IF_NOT_OK` but i don't know if we're allowed to throw exceptions from this context

Reply:
but i like your suggestion. i can just add a log message saying something unexpected happened, and we'll have a way to debug. i'll go with that

Reply:
Fixed here: 02afc7e2aea6c6bb696097f90dc1814188507892

Now we just log an error message